examples/simulation/train.ipynb

```diff
@@ -0,0 +1,289 @@
+    "# get config\n",
+    "cfg = load_config_data(\"./config.yaml\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# rasterisation\n",
+    "rasterizer = build_rasterizer(cfg, dm)\n",
+    "\n",
+    "# ===== INIT DATASET\n",
+    "train_zarr = ChunkedDataset(dm.require(cfg[\"train_data_loader\"][\"key\"])).open()\n",
+    "train_dataset = AgentDataset(cfg, train_zarr, rasterizer)\n",
+    "\n",
+    "# plot some examples\n",
+    "for idx in range(0, len(train_dataset), len(train_dataset) // 10):\n",
+    "    data = train_dataset[idx]\n",
+    "    im = rasterizer.to_rgb(data[\"image\"].transpose(1, 2, 0))\n",
+    "    target_positions = transform_points(data[\"target_positions\"], data[\"raster_from_agent\"])\n",
+    "    draw_trajectory(im, target_positions, TARGET_POINTS_COLOR)\n",
+    "    plt.imshow(im)\n",
+    "    plt.axis('off')\n",
+    "    plt.show()\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model = PlanningModel(\n",
+    "        model_arch=\"resnet50\",\n",
+    "        num_input_channels=rasterizer.num_channels(),\n",
+    "        num_targets=3 * cfg[\"model_params\"][\"future_num_frames\"],  # X, Y, Yaw * number of future states,\n",
+    "        weights_scaling= [1., 1., 1.],\n",
+    "        criterion=nn.MSELoss(reduction=\"none\")\n",
+    "        )\n",
+    "print(model)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Prepare for training\n",
+    "Our `AgentDataset` inherits from PyTorch `Dataset`; so we can use it inside a `Dataloader` to enable multi-processing."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_cfg = cfg[\"train_data_loader\"]\n",
+    "train_dataloader = DataLoader(train_dataset, shuffle=train_cfg[\"shuffle\"], batch_size=train_cfg[\"batch_size\"], \n",
+    "                             num_workers=train_cfg[\"num_workers\"])\n",
+    "device = torch.device(\"cuda:0\" if torch.cuda.is_available() else \"cpu\")\n",
+    "model = model.to(device)\n",
+    "optimizer = optim.Adam(model.parameters(), lr=1e-3)\n",
+    "\n",
+    "print(train_dataset)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Training loop\n",
+    "Here, we purposely include a barebone training loop. Clearly, many more components can be added to enrich logging and improve performance. Still, the sheer size of our dataset ensures that a reasonable performance can be obtained even with this simple loop."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tr_it = iter(train_dataloader)\n",
+    "progress_bar = tqdm(range(cfg[\"train_params\"][\"max_num_steps\"]))\n",
```

Comment:
Any plan to switch to epoch based training?

Reply:
nope, this is just intended as an example for training